helpers/downloader.go

```diff
@@ -37,3 +40,29 @@ func DownloadAndUntar(url, targetDir string) error {
 
 	return UntarStream(resp.Body, targetDir)
 }
+
+// DownloadAndUnzip downloads the archive from given url and decopresses it under the target directory
+func DownloadAndUnzip(url, targetDir string) error {
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	tmpDir, err := ioutil.TempDir("", "fuseml-core")
+	if err != nil {
+		return errors.Wrap(err, "can't create temp directory for download")
+	}
+	defer os.Remove(tmpDir)
+
+	tmpPath := path.Join(tmpDir, "archive.zip")
+	out, err := os.Create(tmpPath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, resp.Body)
+
+	return Unzip(tmpPath, targetDir)
```

Comment:
Instead of using a temporary location to save the file, could you try implementing `DownloadAndUnzip` in the same manner that `DownloadAndUntar` is implemented and use an io.Reader stream ?

Reply:
I don't think so. I tried, and zip library has a different API than the gzip one. It either reads directly from file, or it could read from stream but you'd need to provide a size ... https://golang.org/pkg/archive/zip/#NewReader

Reply:
Too bad though, this would've made for a pretty neat way of transferring the binary straight to disk without the need for an intermediate step :|